src/hotspot/share/opto/machnode.cpp

```diff
@@ -474,14 +474,15 @@ bool MachNode::rematerialize() const {
   }
 
   // Stretching lots of inputs - don't do it.
-  if (req() > 2) {
+  // A MachContant has the last input being the constant base
+  if (req() > (is_MachConstant() ? 3U : 2U)) {
```

Comment:
Saving a spill at the cost of re-materialization using a comparatively cheaper instruction like add/sub/mul looks better for divD may be costly.

Reply:
There are other machine nodes which just accept constants as a mode, like vround* and vcompu* nodes which will now qualify for rematerlization leading to emitting high cost instructions.

Reply:
I think we should have a rough cost model here and not just basing it purely over connectivity of the node, or for the time being you can remove this change ?

Reply:
A node is decided to prefer rematerialising to spilling has to satisfy that:

- The node is not explicitly said to be expensive, `divD` and `divF` fails at this stage.
- The node declaration only contains simple register rules (explicit or implicit DEF dst and USE src), `vround` fails this because it has temp register, `cmpF_imm` and `cmpD_imm` fail this because they kill flags.
- This method we are at agrees with the rematerialising.

I have looked at all instances where `constantaddress` is used and found no node where accidental rematerialisation is inefficient.

Reply:
Thanks for your explanations, I agree.